### PR TITLE
Cut 0.0.113

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

**Comment by arik-so** [comment]
"Workflow completed with no jobs" but there are no merge conflicts. Anything we can do to actually make Github run its actions?

**Comment by arik-so** [comment]
Hm, doesn't appear to have done much.

**Review by arik-so — DISMISSED** [review]

**Review by ViktorTigerstrom — COMMENTED** [review]
Looks good :)! Mostly non blocking suggestions from my part. I realize that my added suggestions would need to take the char limit line break into account if you decide to use them, which they currently don't.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -109,7 +109,7 @@ jobs:
           RUSTFLAGS="-C link-dead-code" cargo build --verbose --color always --features rpc-client,rest-client
           RUSTFLAGS="-C link-dead-code" cargo build --verbose --color always --features rpc-client,rest-client,tokio
       - name: Test backtrace-debug builds on Rust ${{ matrix.toolchain }}
-        if: "matrix.toolchain == stable"
+        if: "matrix.toolchain == 'stable'"
```
Comment (dunxen) [inline .github/workflows/build.yml:112]: Maybe this was also meant to be in the first commit?

**Comment by dunxen** [comment]
LGTM after pending comments.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+ * Onion messages are now read in `CustomOnionMessageHandler` rather than via
+   `MaybeReadableArgs` (#1809).
+ * Added a new util to generate an invoice with a custom hash (#1894) -
+`create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash`
+ * `Sign`ers are now by default re-derived using `KeysInterface`'s new
+   `derive_channel_signer` rather than `read_chan_signer` (#1867).
+ * `Confirm::transactions_confirmed` is now idempotent (#1861).
+ * `ChannelManager::compute_inflight_htlcs` has been added to fetch in-flight
+   HTLCs for scoring, note that `InvoicePayer` does this for you (#1830).
+ * Added the chan we received a payment on to `PaymentReceived` events (#1856).
```
Comment (tnull) [inline CHANGELOG.md:31]:
```suggestion
 * Added the channel over which we received a payment to `PaymentReceived` events (#1856).
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+ * Several bugs in the `lightning-background-processor` `future` feature were
+   fixed, including requirements doc corrections (#1843, #1845, #1851).
+ * Some failure messages sent back when failing an HTLC were corrected (#1895).
+ * `rapid-gossip-sync` no longer errors if an update is applied duplicatively
+   or in rare cases when the graph is updated from payment failures (#1833).
+ * Sending onion msgs when we're the introduction node no longer fails (#1791).
+
+## Backwards Compatibility
+ * No `ChannelReady` events will be generated for previously existing channels,
+   including those which become ready after upgrading 0.0.113 (#1743).
```
Comment (tnull) [inline CHANGELOG.md:57]:
```suggestion
   including those which become ready after upgrading to 0.0.113 (#1743).
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,88 @@
+# 0.0.113 - Dec 15, 2022 - "Big Movement Intercepted"
+
+## API Updates
+ * `ChannelManager::send_payment` now takes an explicit `PaymentId` which is a
+   loose idempotency token. See `send_payment` docs for more (#1761, #1826).
+ * HTLCs bound for SCIDs from `ChannelManager::get_intercept_scid` are now
+   intercepted and can be forwarded manually over any channel (#1835, #1893).
```
[the remark this answers is not in the document]
Comment (dunxen) [reply CHANGELOG.md:7]: Fair point

[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:7]: Ok fair enough :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,88 @@
...
+ * `user_channel_id` is now a `u128` and random for inbound channels (#1790).
+ * A new `ChannelReady` event is generated whenever a channel becomes ready to
+   be used, i.e., after both sides sent the `channel_ready` message (#1743).
+ * `NetworkGraph` now prunes channels where either node is offline for 2 weeks
+   and refuses to accept re-announcements of pruned channels (#1735).
+ * Onion messages are now read in `CustomOnionMessageHandler` rather than via
+   `MaybeReadableArgs` (#1809).
+ * Added a new util to generate an invoice with a custom hash (#1894) -
+`create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash`
+ * `Sign`ers are now by default re-derived using `KeysInterface`'s new
```
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:26]: Ah my bad. Was bit confused given the docs of `derive_channel_signer`.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,88 @@
+# 0.0.113 - Dec 15, 2022 - "Big Movement Intercepted"
+
+## API Updates
+ * `ChannelManager::send_payment` now takes an explicit `PaymentId` which is a
+   loose idempotency token. See `send_payment` docs for more (#1761, #1826).
+ * HTLCs bound for SCIDs from `ChannelManager::get_intercept_scid` are now
+   intercepted and can be forwarded manually over any channel (#1835, #1893).
+ * `Confirm::get_relevant_txids` now returns a `BlockHash`, expanding the set
+   of cases where `transaction_unconfirmed` must be called, see docs (#1796).
```
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:9]: heh fair enough 😁

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,88 @@
...
+   be used, i.e., after both sides sent the `channel_ready` message (#1743).
+ * `NetworkGraph` now prunes channels where either node is offline for 2 weeks
+   and refuses to accept re-announcements of pruned channels (#1735).
+ * Onion messages are now read in `CustomOnionMessageHandler` rather than via
+   `MaybeReadableArgs` (#1809).
+ * Added a new util to generate an invoice with a custom hash (#1894) -
+`create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash`
+ * `Sign`ers are now by default re-derived using `KeysInterface`'s new
+   `derive_channel_signer` rather than `read_chan_signer` (#1867).
+ * `Confirm::transactions_confirmed` is now idempotent (#1861).
```
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:28]: Hmm this was more an attempt to be more consistent with our previous release notes, where we've most of the time used `chain::` to refer to the interfaces in the `chain` module. Though I see one instance in a previous release note where we've used the `Confirm` only. But the other occasions of referring to the different interfaces have used `chain::`. 

Feel free to ignore this though, as I understand that you may want to prioritize keeping the lines short :).

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,88 @@
...
+   fixed, including requirements doc corrections (#1843, #1845, #1851).
+ * Some failure messages sent back when failing an HTLC were corrected (#1895).
+ * `rapid-gossip-sync` no longer errors if an update is applied duplicatively
+   or in rare cases when the graph is updated from payment failures (#1833).
+ * Sending onion msgs when we're the introduction node no longer fails (#1791).
+
+## Backwards Compatibility
+ * No `ChannelReady` events will be generated for previously existing channels,
+   including those which become ready after upgrading 0.0.113 (#1743).
+ * Once `UserConfig::accept_intercept_htlcs` is set, downgrades to LDK versions
```
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:59]: Fair enough :)

**Review by arik-so — APPROVED** [review]

**Comment by arik-so** [comment]
Could you squash?

File: CHANGELOG.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply CHANGELOG.md:28]: Yeah, let's keep it as it is then :)

**Review by ViktorTigerstrom — APPROVED** [review]

**Review by arik-so — APPROVED** [review]

**Pull request merged**